firefox-ios/Client/Frontend/Browser/Toolbars/Redux/ToolbarState.swift

```diff
@@ -105,217 +105,257 @@ struct ToolbarState: ScreenState, Equatable {
 
         switch action.actionType {
         case ToolbarActionType.didLoadToolbars:
-            guard let toolbarAction = action as? ToolbarAction,
-                  let toolbarPosition = toolbarAction.toolbarPosition
-            else { return state }
+            return handleDidLoadToolbars(state: state, action: action)
 
-            let position = addressToolbarPositionFromSearchBarPosition(toolbarPosition)
-            return ToolbarState(
-                windowUUID: state.windowUUID,
-                toolbarPosition: position,
-                isPrivateMode: state.isPrivateMode,
-                addressToolbar: AddressBarState.reducer(state.addressToolbar, toolbarAction),
-                navigationToolbar: NavigationBarState.reducer(state.navigationToolbar, toolbarAction),
-                isShowingNavigationToolbar: state.isShowingNavigationToolbar,
-                isShowingTopTabs: state.isShowingTopTabs,
-                canGoBack: state.canGoBack,
-                canGoForward: state.canGoForward,
-                numberOfTabs: state.numberOfTabs,
-                showMenuWarningBadge: state.showMenuWarningBadge,
-                isNewTabFeatureEnabled: toolbarAction.isNewTabFeatureEnabled ?? state.isNewTabFeatureEnabled,
-                canShowDataClearanceAction: toolbarAction.canShowDataClearanceAction ?? state.canShowDataClearanceAction,
-                canShowNavigationHint: state.canShowNavigationHint)
-
-        case ToolbarActionType.borderPositionChanged,
-            ToolbarActionType.urlDidChange,
-            ToolbarActionType.didSetTextInLocationView,
-            ToolbarActionType.didPasteSearchTerm,
-            ToolbarActionType.didStartEditingUrl,
-            ToolbarActionType.cancelEdit,
-            ToolbarActionType.didScrollDuringEdit,
-            ToolbarActionType.websiteLoadingStateDidChange,
+        case ToolbarActionType.borderPositionChanged, ToolbarActionType.urlDidChange,
+            ToolbarActionType.didSetTextInLocationView, ToolbarActionType.didPasteSearchTerm,
+            ToolbarActionType.didStartEditingUrl, ToolbarActionType.cancelEdit,
+            ToolbarActionType.didScrollDuringEdit, ToolbarActionType.websiteLoadingStateDidChange,
```

Comment:
I don't have a problem with this. CC @thatswinnie.

Reply:
I could have sworn Swiftlint would complain about this but it's actually fine. However I find this harder to read. What's the benefit of changing it like this?

Reply:
AFAIK, Swiftlint rule `closure_body_length` is set to 30 lines. Without this change in the `reducer` Swiftlint would still complain, right @ionixjunior?

Reply:
Exactly, @PARAIPAN9. Without these changes, the `reducer` will be 33 lines, which triggers a warning from the linter since the limit is 30 lines.

Honestly, I agree that this impacts readability, as @thatswinnie also pointed out. This solution addresses the current issue, but it could make it easier to exceed the limit again in the future if more cases are added to the switch. I'm open to any suggestions for improving this further.